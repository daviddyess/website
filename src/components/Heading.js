import PropTypes from 'prop-types';
import classNames from 'classnames';
import React, { Component } from 'react';

import './Heading.scss';

export const modifiers = ['primary'];

export default class Heading extends Component {
  static propTypes = {
    className: PropTypes.string,
    element: PropTypes.string,
    modifier: PropTypes.oneOf(modifiers),
    children: PropTypes.node.isRequired
  };

  static defaultProps = {
    element: 'h1'
  };

  render() {
    const { element: Element, modifier, children, className } = this.props;

    const mainClass = 'gp-heading';
    const modifierClass = modifier ? `${mainClass}--${modifier}` : false;

    return (
      <Element className={classNames(mainClass, modifierClass, className)}>
        {children}
      </Element>
    );
  }
}
