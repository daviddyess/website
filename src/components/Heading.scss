@import "../../node_modules/bootstrap/scss/bootstrap-grid";

.gp-heading--primary {
  @include media-breakpoint-up(lg) {
    padding-bottom: 40px;
  }
}
