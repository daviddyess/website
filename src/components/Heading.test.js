import React from 'react';
import renderer from 'react-test-renderer';

import Heading from './Heading';

describe('Heading', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Heading>Test</Heading>).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders with custom element', () => {
    const tree = renderer.create(<Heading element="h5">Test</Heading>).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders with modifier', () => {
    const tree = renderer
      .create(<Heading modifier="primary">Test</Heading>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
