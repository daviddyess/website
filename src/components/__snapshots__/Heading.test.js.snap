// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading renders correctly 1`] = `
<h1
  className="gp-heading"
>
  Test
</h1>
`;

exports[`Heading renders with custom element 1`] = `
<h5
  className="gp-heading"
>
  Test
</h5>
`;

exports[`Heading renders with modifier 1`] = `
<h1
  className="gp-heading gp-heading--primary"
>
  Test
</h1>
`;
